hurumap/management/commands/importdata.py

```diff
@@ -0,0 +1,158 @@
+        )
+
+    def debug(self, msg):
+        if self.verbosity >= 2:
+            self.stdout.write(str(msg))
+
+    def handle(self, *args, **options):
+        self.filepath = options['filepath']
+
+        self.verbosity = options.get('verbosity', 1)
+        self.table_id = options.get('table')
+        self.geo_version = options.get('geo_version')
+        self.columns = options.get('columns', [])
+        self.value_type = options.get('value_type', 'Integer')
+        self.dryrun = options.get('dryrun', False)
+
+        if self.dryrun:
+            self.stdout.write("DRY RUN: not actually writing data")
+
+        with open(self.filepath) as f:
+            self.f = f
+            self.reader = csv.DictReader(self.f, delimiter=",")
+            # Fields excluding geo_level, geo_code and total
+            self.fields = self.reader.fieldnames[2:-1]
+
+            self.setup_table()
+            self.store_values()
+
+    def setup_table(self):
+        table_id = self.table_id or get_table_id(self.fields)
+        try:
+            self.table = get_datatable(table_id)
+            self.stdout.write(
+                "Table for fields %s is %s" %
+                (self.fields, self.table.id))
+        except KeyError:
+            raise CommandError(
+                "Couldn't establish which table to use for these fields. "
+                "Have you added a FieldTable entry in wazimap_za/tables.py?\nFields: %s" %
+                self.fields)
+
+    def store_values(self):
+        session = get_session()
+        count = 0
+        for row in self.reader:
+            model_row = {}
+            count += 1
+            geo_code, geo_level = self.get_geo_data(row['geography'])
+            model_row['geo_version'] = self.geo_version
+            model_row['geo_code'] = geo_code
+            model_row['geo_level'] = geo_level
+
+            for col in self.columns:
+                model_row[col] = row[col]
+
+            if row['total'] == 'no data':
+                model_row['total'] = None
+            else:
+                model_row['total'] = round(
+                    float(
+                        row['total']),
+                    1) if self.value_type == 'Float' else int(
+                    round(
+                        float(
+                            row['total'])))
+            # self.stdout.write("%s-%s" % (row['geo_level'], row['geo_code'])
+            print model_row
+            entry = self.table.model(**model_row)
+
+            if not self.dryrun:
+                session.add(entry)
+
+            if count % 100 == 0:
+                session.flush()
+
+        if not self.dryrun:
+            session.commit()
+
+        session.close()
+
+    def get_geo_data(self, geo_name):
+        try:
+            query = Geography.objects.get(name__iexact=geo_name)
+            return query.geo_code, query.geo_level
+        except ObjectDoesNotExist:
+            raise CommandError(geo_name + " does not exist")
+
```

Comment:
What's the difference between this and `importcsv`?

Reply:
`importcsv` is customized to work with South African geography, `importdata` will work for any   country and whatever geography structure they use provided the wazimap_geography.sql has already been added to the DB

Reply:
That's awesome. Let's replace `importcsv` with this code then.